### ISSUE TITLE
separating cn/xtl from cn/1 in algo_perf

### ISSUE THREAD

**Issue by tinsami1** [MoneroOcean/meta-miner#4]
Hi!

Since claymore do not support cn/1 variants (and no or buggy support from xmr-stak, as well), I'm looking for a way to detach cn/xtl (and xao and rto) from cn/1's algo_perf value so that I can point it to xmrig instead.  So I added the following in mm.js:

```
const algo_perf_algo = {
  "cn":       "cn/1",
  "cn-xtl":   "cn/xtl",          // <== Added this here...
  "cn-fast":  "cn/msr",
  "cn-lite":  "cn-lite/1",
  "cn-heavy": "cn-heavy/0",
};

function algo_perf_class(algo) { // converts algo to algo class
   if (algo.indexOf("heavy") > -1) return "cn-heavy";
   if (algo.indexOf("lite")  > -1) return "cn-lite";
   if (algo.indexOf("msr")   > -1) return "cn-fast";
   if (algo.indexOf("rto")   > -1) return "cn-xtl";       // <== and this,
   if (algo.indexOf("xao")   > -1) return "cn-xtl";       // <== and this,
   if (algo.indexOf("xtl")   > -1) return "cn-xtl";       // <== and this here.
   return "cn";
```

Then I manually added an algo_perf entry in mm.json for cn/xtl

Is that enough?

**Comment by MoneroOcean** [comment]
Unfortunately this will be not supported by the pool. It just do not make difference between performance of cryptonight variants (XTL, RTO, ...), so changes in mm.js will be useless here. How different cn/xtl claymore performance from xmr-stak cn? If it is couple of percents then it does not matter much. If difference is greater then I guess one of the miners is not optimally configured.

**Comment by tinsami1** [comment]
The performance difference is >30%.

In a Win10 rig with 3 HD 78xx cards and using Adrenalin drivers, I get 1200Hs with claymore.  The best I can get with xmrig or xmr-stak is close to 900Hs -- and I've even tried various driver versions (Adrenaline, Crimson, Blockchain, even old Catalyst).  Even srbminer is only at par with xmrig and xmr-stak.

It is not a big deal.  I can live with not mining XTL on my HD78xxx cards.  Such a shame that there won't be any new Claymore CN miner.

**Comment by MoneroOcean** [comment]
US/JP nodes already use more precise hashrate detection (DE/FR will join them after next reboot), so you can specify cn/xtl algo-perf hashrate manually in mm.json so it will be taken into account.

**Comment by tinsami1** [comment]
Cool!!!  I'll test it out within the next couple of days.  I have an all-AMD rig that is down for maintenance right now.  I do the test with it once I have it back up and running.

**Comment by tinsami1** [comment]
Ok.  This works.  I set up one of my cpu miners to mine XTL only if it is +10% of XMR.  I monitored it for a while.  During that time XTL just wasn't going past the 110% and mm didn't switch to it.  Left it alone for a while, then when I came back I saw that it switched to XTL sometime before I came back and then back to cn/1 -- I haven't tracked if XTL went past 110%, just assumed that it was 'coz it was a time pool mined a number of XTL blocks, so I figure there's a good chance it did indeed go beyond and that mm switched because of it.
